components/DataSchema.tsx

```diff
@@ -134,63 +143,165 @@ function getColumnDef(dataSchemaMap?: {
     };
 }
 
-const DataSchemaTable: React.FunctionComponent<{
+interface DataSchemaTableProps {
     schemaData: DataSchemaData[];
     dataSchemaMap?: { [id: string]: DataSchemaData };
     title?: string;
     columns?: ColumnName[];
-}> = (props) => {
-    // include Attribute and Description columns by default
-    // (exclude Label and Required columns by default)
-    const availableColumns = props.columns || [
-        ColumnName.Attribute,
-        ColumnName.Description,
-    ];
-    // include Valid Values column only if there is data
-    if (
-        !availableColumns.includes(ColumnName.ValidValues) &&
-        hasNonEmptyValidValues(props.schemaData)
-    ) {
-        availableColumns.push(ColumnName.ValidValues);
+    filterControl: DataStandardFilterControl;
+}
+
+const DataSchemaTable: React.FunctionComponent<DataSchemaTableProps> = observer(
+    (props) => {
+        const availableColumns = props.columns || [
+            ColumnName.Attribute,
+            ColumnName.Description,
+            ColumnName.ValidValues,
+        ];
+
+        const columnDef = getColumnDef(props.dataSchemaMap);
+        const columns: IDataTableColumn[] = _.uniq(availableColumns).map(
+            (name) => columnDef[name]
+        );
+
+        const filteredSchemaDataById =
```

Comment:
I've built the search on top of the filter, so when an individual searches for an attribute then that attribute from filters is selected and the result is displayed, thus I kept this logic because for filtering and displaying data, it's required.